### PR TITLE
[FLINK-28090][table] Support attachAsDataStream

### PR DESCRIPTION
**Pull request by billyrrr**
## What is the purpose of the change
Implement attachAsDataStream

## Brief change log

- support attachAsDataStream

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): don't know
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? not documented

### CODE REVIEW

File: flink-python/pyflink/table/statement_set.py [hunk 1]
```diff
@@ -50,6 +50,13 @@ def add_insert_sql(self, stmt: str) -> 'StatementSet':
         """
         self._j_statement_set.addInsertSql(stmt)
         return self
+    
+    def attach_as_datastream(self: StatementSet):
```
Comment (dianfu) [inline flink-python/pyflink/table/statement_set.py:54]: Could you also update the documentation here: https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/data_stream_api/#adding-table-api-pipelines-to-datastream-api

File: flink-python/pyflink/table/statement_set.py [hunk 2]
```diff
@@ -50,6 +50,13 @@ def add_insert_sql(self, stmt: str) -> 'StatementSet':
...
+        """
+        combine table api statements into datastream
```
Comment (dianfu) [inline flink-python/pyflink/table/statement_set.py:56]: Could we reference the documentation from the Java attachAsDataStream?

File: flink-python/pyflink/table/statement_set.py (hunk 1 above)
Comment (billyrrr) [reply flink-python/pyflink/table/statement_set.py:54]: Definitely

File: flink-python/pyflink/table/statement_set.py (hunk 2 above)
Comment (billyrrr) [reply flink-python/pyflink/table/statement_set.py:56]: I'll do that and adapt the docs to the python

Comment (dianfu) [reply flink-python/pyflink/table/statement_set.py:56]: @billyrrr Thanks. Besides, there are test failures(check style issues): https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=36800&view=logs&j=9cada3cb-c1d3-5621-16da-0f718fb86602&t=c67e71ed-6451-5d26-8920-5a8cf9651901. Please also take a look when updating the PR~

**Comment by dianfu** [comment]
@billyrrr What's the status of this PR?

**Pull request closed without merging**